### PR TITLE
Specify output folder for each download and add the option to disable file name checking

### PR DESCRIPTION
**Pull request by ALEZ-DEV**
So I added two thing in this PR:
1. Can now specify an output folder for each download

We can now create a new `Download` instance with `Download::new_with_output` and specify the `url` and the `output_path`  
like this :

```rust
let file1 = Download::new_with_output("https://example.com/file1", "/home/someuser/Documents");
let file2 = Download::new("https://example.com/file2")

let mut downloader = Downloader::builder()
            .download_folder("/home/someuser/Downloads")
            .parallel_requests(5)
            .build().unwrap();

downloader.download(&[file1, file2]);

// the file1 will be write in /home/someuser/Documents
// the file2 will be write in /home/someuser/Downloads
```

as you can see, `Download` instance initialized with `Download::new` will be only be affected by the default download_folder

2. Can now disable de file name checking if needed

If needed, we can now disable the check of the file name for each download, why I did this ? because I'm downloading a lot of files and some of them have the same name but are not downloaded in the same path.

little example: 
```rust
let mut file1 = Download::new("https://example.com/file1");
file1.check_file_name = false; // this will disable the name checking
```

I know that's not the best implementation, but if you want I can refactor it
If you have some idea, please give me your thought!

### CODE REVIEW

**Comment by ALEZ-DEV** [comment]
And btw, if this PR is merged, can you publish the new version on crates.io? Thanks in advance!

**Comment by ALEZ-DEV** [comment]
In that case, if we can't canonicalize the path, can't we force the use of a complete path instead of a relative one ? Even if that break the current api for certain people ? like check the use of `/` or `C:` at the start of the path, even if this sound not like the best idea.

but a was thinking in another way to do this, if `Download.output_path` is `Some(path)`, we just disable the file checking, I think there is a better way to do this but I don't know, what do you think about that ?

File: src/download.rs [hunk 1]
```diff
@@ -17,6 +17,10 @@ pub struct Download {
     pub progress: Option<crate::Progress>,
     /// The file name to be used for the downloaded file.
     pub file_name: std::path::PathBuf,
+    /// If set to `true` will check if there are a same file name and throw an error if that's the case
+    pub check_file_name: bool,
+    /// The output path used to write the file
+    pub output_path: Option<std::path::PathBuf>,
```
[the remark this answers is not in the document]
Comment (ALEZ-DEV) [reply src/download.rs:23]: You mean deleting `check_file_name` and `output_path` ?
I think that's okay, but we have to keep the `Download::new_with_output`, but instead to set the file path in the `output_path`, we set the `file_name`, that's what you mean ?

**Comment by hunger** [comment]
Closing as already implemented ;-)

**Pull request closed without merging**